File: Livro - Módulo Java/3 - Estrutura de Decisão/Exercicio.17.java
//Desenvolva um algoritmo para calcular quantos reais serão necessários para encher o tanque de um veículo para se realizar uma viagem. O usuário deverá informar o tipo de combustível do veículo, o número total de km a ser percorrido e o consumo médio do veículo. A tabela de preços dos combustíveis utilizada no cálculo é apresentada abaixo

// Combustível 	Preço 
// Gasolina 	R$ 22,25 
// Álcool 	R$ 11,50 
// Diesel 	R$ 11,65
import java.util.Scanner;

public class Exercicio17 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Informe o tipo de combustível (gasolina, álcool ou diesel): ");
        String tipoCombustivel = scanner.nextLine().toLowerCase();

        System.out.print("Informe o número total de km a ser percorrido: ");
        double kmPercorridos = scanner.nextDouble();

        System.out.print("Informe o consumo médio do veículo em km por litro: ");
        double consumoMedio = scanner.nextDouble();

        System.out.println(calcularCustoCombustivel(tipoCombustivel, kmPercorridos, consumoMedio));

        scanner.close();
    }

    private static String calcularCustoCombustivel(String tipoCombustivel, double kmPercorridos, double consumoMedio) {
        final double PRECO_GASOLINA = 22.25;
        final double PRECO_ALCOOL = 11.50;
        final double PRECO_DIESEL = 11.65;

        double precoPorLitro = 0.0;

        switch (tipoCombustivel) {
            case "gasolina":
                precoPorLitro = PRECO_GASOLINA;
                break;
            case "álcool":
                precoPorLitro = PRECO_ALCOOL;
                break;
            case "diesel":
                precoPorLitro = PRECO_DIESEL;
                break;
            default:
                return "Tipo de combustível inválido. Por favor, escolha entre gasolina, álcool ou diesel.";
        }

        double litrosNecessarios = kmPercorridos / consumoMedio;
        double custoTotal = litrosNecessarios * precoPorLitro;

        return String.format("Para percorrer %.2f km, o custo de combustível será de R$ %.2f.", kmPercorridos, custoTotal);
    }
}

File: Livro - Módulo Python/3 - Estrutura de Decisão/Exercicio.17.py
#17. Desenvolva um algoritmo para calcular quantos reais serão necessários para encher o tanque de um veículo para se realizar
# uma viagem. O usuário deverá informar o tipo de combustível do veículo, o número total de km a ser percorrido e o consumo médio
# do veículo. A tabela de preços dos combustíveis utilizada no cálculo é apresentada abaixo: 
# Combustível - Preço 
# Gasolina - R$ 22,25 
# Álcool - R$ 11,50 
# Diesel - R$ 11,65 

combustivel = input("Digite o tipo de combustível do veículo (Gasolina, Álcool ou Diesel): ")
km_percorridos = float(input("Digite o número total de km a serem percorridos: "))
consumo_medio = float(input("Digite o consumo médio do veículo (em km/l): "))

preco_gasolina = 22.25
preco_alcool = 11.5
preco_diesel = 11.65

valor_total = 0.0

if combustivel.lower() == "gasolina":
    valor_total = (km_percorridos / consumo_medio) * preco_gasolina
if combustivel.lower() == "álcool":
    valor_total = (km_percorridos / consumo_medio) * preco_alcool
if combustivel.lower() == "diesel":
    valor_total = (km_percorridos / consumo_medio) * preco_diesel

print("Valor total necessário para encher o tanque: R$", valor_total)